core/src/main/scala/slamdata/AutoBump.scala

```diff
@@ -0,0 +1,369 @@
+
+  def autoBumpCommitTitle(author: String) = s"Applied dependency updates by $author"
+
+  def autoPage[F[_]: Sync, T](first: Pagination)
+                             (call: Pagination => F[Either[GHException, GHResult[List[T]]]]): Stream[F, T] = {
+    val chunker = call.andThen(_.rethrow.map(res => nextPage(getRelations(res.headers)).map(Chunk.seq(res.result) -> _)))
+    Stream.unfoldChunkEval(first)(chunker)
+  }
+
+  def nextPage(relations: Map[String, (Int, Int)]): Option[Pagination] = {
+    relations.get("next").map((Pagination.apply _).tupled)
+  }
+
+  /** Decodes github's "Link" header into a map */
+  def getRelations(headers: Map[String, String]): Map[String, (Int, Int)] = {
+    val relations = for {
+      header <- headers.map { case (k, v) => k.toLowerCase -> v }.collect { case ("link", header) => header }
+      LinkRelation(url, relation) <- LinkRelation.findAllMatchIn(header)
+      uri <- Uri.fromString(url).toSeq
+      page <- uri.params.get("page")
+      pageNum <- Try(page.toInt).toOption
+      perPage <- uri.params.get("per_page")
+      perPageNum <- Try(perPage.toInt).toOption
+    } yield (relation, (pageNum, perPageNum))
+    relations.toMap
+  }
+
+  /** Extract change label from trickleUpdateDependencies log */
+  def extractLabel(lines: List[String]): ChangeLabel = {
+    lines.collectFirst {
+      case ChangeLabel(label) => label
+    }.getOrElse(sys.error("Change label not found!"))
+  }
+
+  /** Extract updated versions from trickleUpdateDependencies log */
+  def extractChanges(lines: List[String]): List[String] = {
+    lines.filter(_ contains "Updated ").map(line => line.substring(line.indexOf("Updated ")))
+  }
+
+  def getBranch(pullRequest: Option[PullRequestDraft]): IO[(String, String)] = IO {
+    pullRequest
+      .map("" -> _.head.get.ref)
+      .getOrElse("-b" -> s"trickle/version-bump-${System.currentTimeMillis()}")
+  }
+
+  def isAutoBump(pullRequest: PullRequestDraft, labels: List[Label]): Boolean = {
+    pullRequest.head.exists(_.ref.startsWith("trickle/")) && labels.exists(_.name == AutoBumpLabel)
+  }
+}
+
+class AutoBump(repository: OutdatedRepository, token: String) {
+  import AutoBump._
+
+  assert(url(repository.url).getHost == "github.com")
+
+  implicit private val IOContextShift: ContextShift[IO] = contextShift(global)
+  val github: Github[IO] = Github[IO](Some(token))
+  val (owner, repoSlug) = repository.ownerAndRepository.getOrElse(sys.error(s"invalid url ${repository.url}"))
+  val sbt: String = sys.env.getOrElse("SBT", "sbt")
+
+  /**
+   * Finds primary open autobump pull request, if one exists.
+   *
+   * @return Oldest autobump pull request, by order of creation
+   */
+  def getOldestAutoBumpPullRequest: IO[Option[PullRequestDraft]] = {
+    getPullRequests
+      .evalFilter(pullRequest => getLabels(pullRequest.number).map(labels => isAutoBump(pullRequest, labels)))
+      .head
+      .compile.toList
+      .map(_.headOption)
```

Comment:
```suggestion
      .compile.last
```

Reply:
Last? Steam is returning a list of prs <<pr1, pr2, pr3, ..., prn>>, where pr1 is the oldest and prn is the newest, and I want the oldest, which is what the code was doing. `last` seems to do very much the opposite,no?